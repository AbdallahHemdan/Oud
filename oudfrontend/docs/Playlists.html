<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Playlists</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Playlists</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Playlists<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">this is a component that renders users favourite and created playlists</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="Playlists"><span class="type-signature"></span>new Playlists<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>state</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last">carries the state of the component
                <h6>Properties</h6>

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>recieved</code></td>
            

            <td class="type">
            
                
<span class="param-type">boolean</span>


            
            </td>

            

            

            <td class="description last">it is false when the playlists are fetched successfully Otherwise, it carries the decode id</td>
        </tr>

    

        <tr>
            
                <td class="name"><code>playlists</code></td>
            

            <td class="type">
            
                
<span class="param-type">object</span>


            
            </td>

            

            

            <td class="description last">carries the users favourite and created playlists</td>
        </tr>

    
    </tbody>
</table>

            </td>
        </tr>

    
    </tbody>
</table>




<dl class="details">

    

    

    

    

    

    

    

    

    
    <dt class="tag-author">Author:</dt>
    <dd class="tag-author">
        <ul>
            <li><a href="mailto:ahmedwa1999@gmail.com">Ahmed Walid</a></li>
        </ul>
    </dd>
    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_library_components_playlists_playlists.jsx.html">components/library/components/playlists/playlists.jsx</a>, <a href="components_library_components_playlists_playlists.jsx.html#line26">line 26</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    {      <div>       <h1></h1> Followed Playlists </h1>                {this.state.recieved?          <CategoryBody  playlists={this.state.playlists}/>          :<LoadingSnipper/>          }      </div>
</div>



    





    
    </div>

    

    

    
        <h3 class="subsection-title">Classes</h3>

        <dl>
            <dt><a href="Playlists.html">Playlists</a></dt>
            <dd></dd>
        </dl>
    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="AccountOverview.html">AccountOverview</a></li><li><a href="addToPlaylist.html">addToPlaylist</a></li><li><a href="Album.html">Album</a></li><li><a href="Albums.html">Albums</a></li><li><a href="ChangePassword.html">ChangePassword</a></li><li><a href="CreatePlaylist.html">CreatePlaylist</a></li><li><a href="EditProfile.html">EditProfile</a></li><li><a href="Facebook.html">Facebook</a></li><li><a href="FollowCard.html">FollowCard</a></li><li><a href="Followers.html">Followers</a></li><li><a href="Following.html">Following</a></li><li><a href="ForgotPassword.html">ForgotPassword</a></li><li><a href="GenreCard.html">GenreCard</a></li><li><a href="HeaderBody.html">HeaderBody</a></li><li><a href="HeaderBodyTop.html">HeaderBodyTop</a></li><li><a href="Home.html">Home</a></li><li><a href="Library.html">Library</a></li><li><a href="LikedSongs.html">LikedSongs</a></li><li><a href="MainBrand.html">MainBrand</a></li><li><a href="MainContent.html">MainContent</a></li><li><a href="MusicCard.html">MusicCard</a></li><li><a href="MusicItem.html">MusicItem</a></li><li><a href="Navbar.html">Navbar</a></li><li><a href="Player.html">Player</a></li><li><a href="PlayingBarRight.html">PlayingBarRight</a></li><li><a href="Playlist.html">Playlist</a></li><li><a href="Playlists.html">Playlists</a></li><li><a href="PublicPlaylists.html">PublicPlaylists</a></li><li><a href="Queue.html">Queue</a></li><li><a href="ResetPassword.html">ResetPassword</a></li><li><a href="SideBar.html">SideBar</a></li><li><a href="Sidebar_.html">Sidebar</a></li><li><a href="SignIn.html">SignIn</a></li><li><a href="Signup.html">Signup</a></li><li><a href="SocialIcons.html">SocialIcons</a></li><li><a href="Song.html">Song</a></li><li><a href="SongList.html">SongList</a></li><li><a href="Track.html">Track</a></li><li><a href="UpperContainer.html">UpperContainer</a></li><li><a href="UserExperienceForm.html">UserExperienceForm</a></li><li><a href="WebPlayer.html">WebPlayer</a></li></ul><h3>Global</h3><ul><li><a href="global.html#addToQueue">addToQueue</a></li><li><a href="global.html#BeforeLogin">BeforeLogin</a></li><li><a href="global.html#CardList">CardList</a></li><li><a href="global.html#checkPassword">checkPassword</a></li><li><a href="global.html#EditCountry">EditCountry</a></li><li><a href="global.html#EditGendr">EditGendr</a></li><li><a href="global.html#EditProfileTextElement">EditProfileTextElement</a></li><li><a href="global.html#FeedBack">FeedBack</a></li><li><a href="global.html#fetchCategoriesUrl">fetchCategoriesUrl</a></li><li><a href="global.html#Footer">Footer</a></li><li><a href="global.html#formValid">formValid</a></li><li><a href="global.html#HeaderBodyBottom">HeaderBodyBottom</a></li><li><a href="global.html#HelpUs">HelpUs</a></li><li><a href="global.html#HiddenSideBar">HiddenSideBar</a></li><li><a href="global.html#ImageContainer">ImageContainer</a></li><li><a href="global.html#LowerContainer">LowerContainer</a></li><li><a href="global.html#MainContainer">MainContainer</a></li><li><a href="global.html#Overview">Overview</a></li><li><a href="global.html#PageContainer">PageContainer</a></li><li><a href="global.html#pause">pause</a></li><li><a href="global.html#PlayingBarCenter">PlayingBarCenter</a></li><li><a href="global.html#PlayingBarLeft">PlayingBarLeft</a></li><li><a href="global.html#Profile">Profile</a></li><li><a href="global.html#ProfileElement">ProfileElement</a></li><li><a href="global.html#ProfileInfo">ProfileInfo</a></li><li><a href="global.html#render">render</a></li><li><a href="global.html#resume">resume</a></li><li><a href="global.html#SideBarElement">SideBarElement</a></li><li><a href="global.html#SignOutEverywhere">SignOutEverywhere</a></li><li><a href="global.html#SortableTrackContainer">SortableTrackContainer</a></li><li><a href="global.html#ThankYou">ThankYou</a></li><li><a href="global.html#TrackContainer">TrackContainer</a></li><li><a href="global.html#YourPlan">YourPlan</a></li><li>{JSX}</li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.3</a> on Wed Apr 22 2020 21:05:29 GMT+0200 (Eastern European Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>