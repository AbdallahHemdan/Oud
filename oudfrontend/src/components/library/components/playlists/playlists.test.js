import React from 'react'
import Playlists from './playlists.jsx';
import Enzyme, { shallow } from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new EnzymeAdapter() });


const setup = () => {
    return shallow(<Playlists />);
}

const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-testid="${val}"]`);
}
/*
Rendering tests
*/

describe('playlists Component', () => {
    describe('testing library playlists renders Correctly', () => {
        let component;
        beforeEach(() => {
            component = setup();
        })
        it('renders playlists component', () => {
            const wrapper = findByTestAttr(component, "playlists");
            expect(wrapper.length).toBe(1);
        });

        it('renders title component', () => {
            const wrapper = findByTestAttr(component, "title");
            expect(wrapper.length).toBe(1);
            expect(wrapper.text()).toBe(' Followed Playlists ')
        });
        it('renders 1 card component', () => {
            component.setState({ recieved: true, playlists: [1] })
            const wrapper = findByTestAttr(component, "categoryBody");
            expect(wrapper.length).toBe(1);
        });
        it('renders no cards component when recieved is false', () => {
            component.setState({ playlists: [1, 1], recieved: false })
            const wrapper = findByTestAttr(component, "categoryBody");
            expect(wrapper.length).toBe(0);
        });
        it('does not render loading component when recieved is true', () => {
            component.setState({ recieved: true });
            const wrapper = findByTestAttr(component, "loading");
            expect(wrapper.length).toBe(0);
        });
        it('renders loading component when recieved is false', () => {
            component.setState({ recieved: false });
            const wrapper = findByTestAttr(component, "loading");
            expect(wrapper.length).toBe(1);
        });

    });
    describe('snapshot test', () => {
        it('renders correctly', () => {
            const tree = renderer
                .create(<Playlists />)
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});