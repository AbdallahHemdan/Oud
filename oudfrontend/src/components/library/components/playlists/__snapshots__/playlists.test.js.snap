// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playlists Component snapshot test renders correctly 1`] = `
<div
  data-testid="playlists"
>
  <h1
    data-testid="title"
  >
     Followed Playlists 
  </h1>
  <div
    className="spinner"
    data-testid="snipper"
  >
    <div
      className="rect1"
      data-testid="snipper-1"
    />
    <div
      className="rect2"
      data-testid="snipper-2"
    />
    <div
      className="rect3"
      data-testid="snipper-3"
    />
    <div
      className="rect4"
      data-testid="snipper-4"
    />
    <div
      className="rect5"
      data-testid="snipper-5"
    />
  </div>
</div>
`;
